### PR TITLE
task/kernel.py: fix -k distro for opensuse/sle (part 1)

### PR DESCRIPTION
**Pull request by smithfarm**
Part one - this does not get us all the way, but it's a start!

Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

File: teuthology/task/kernel.py [hunk 1]
```diff
@@ -721,37 +722,44 @@ def need_to_install_distro(remote):
...
-                        'Newest distro kernel already installed and running')
-                    return False
-            else:
-                remote.run(args=['sudo', 'yum', 'reinstall', '-y', 'kernel',
-                                 run.Raw('||'), 'true'])
+        if dist_release in ['opensuse', 'sle']:
+            install_stdout = remote.sh(
+                'sudo zypper --non-interactive install kernel-default'
+                )
+            match = re.search(
```
Comment (toabctl) [inline teuthology/task/kernel.py]: not needed because not used in the sle/opensuse case

File: teuthology/task/kernel.py [hunk 2]
```diff
@@ -1040,7 +1058,7 @@ def get_latest_image_version_rpm(remote):
     for kernel in proc.stdout.getvalue().split():
         if kernel.startswith('kernel'):
             if 'ceph' not in kernel:
-                version = kernel.split('kernel-')[1]
+                version = kernel.split('{}-'.format(kernel_pkg_name))[1]
```
Comment (kshtsk) [inline teuthology/task/kernel.py]: Using 'format` here is overkill.

File: teuthology/task/kernel.py (hunk 1 above)
Comment (kshtsk) [reply teuthology/task/kernel.py]: agree, looks like it

**Comment by kshtsk** [comment]
I guess this PR is not ready:
```
teuthology/task/kernel.py:844:24: F821 undefined name 'distribution'
teuthology/task/kernel.py:873:24: F821 undefined name 'distribution'
```

**Comment by smithfarm** [comment]
@toabctl @kshtsk Thanks for reviewing. Please take another look.

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by smithfarm** [comment]
@toabctl Has your change request been addressed?

**Pull request merged**